Bot/Program.cs

```diff
@@ -1,33 +1,74 @@
+using Bot.Discord.Handlers;
 using Bot.Discord.Options;
+using Bot.Internal.Events;
 using Bot.Services;
+using Bot.WebApi.Options;
+using Discord;
+using Discord.Interactions;
+using Discord.WebSocket;
 
-var builder = WebApplication.CreateBuilder(args);
-
-// Add services to the container.
-builder.Services.AddRazorPages();
-
-builder.Services.AddLogging();
+namespace Bot
+{
+	public class Program
+	{
+        public static void Main(string[] args) { ConfigureApp(ConfigureServices(WebApplication.CreateBuilder(args))).Run(); }
 
-builder.Services.Configure<DiscordConfiguration>(builder.Configuration.GetSection("DiscordSettings"));
-builder.Services.AddHostedService<DiscordBotService>();
+        private static WebApplication ConfigureServices(WebApplicationBuilder builder)
+		{
+            builder.Services.AddRazorPages();
+            builder.Services.AddLogging(log =>
+            {
+                log.ClearProviders();
+                log.AddSimpleConsole(cnf => { cnf.TimestampFormat = "[MM-dd-yyyy HH:mm:ss:fff] "; });
+            });
 
-var app = builder.Build();
+            // Configurations
+            builder.Services.Configure<DiscordConfiguration>(builder.Configuration.GetSection("DiscordSettings"));
+            builder.Services.Configure<ApiConfiguration>(builder.Configuration.GetSection("ApiConfiguration"));
 
-// Configure the HTTP request pipeline.
-if (!app.Environment.IsDevelopment())
-{
-	app.UseExceptionHandler("/Error");
-	// The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
-	app.UseHsts();
-}
+            // Discord client
+            builder.Services.AddSingleton(_ =>
+            {
+                var client = new DiscordSocketClient(new DiscordSocketConfig
+                {
+                    AlwaysDownloadUsers = true,
+                    GatewayIntents = GatewayIntents.AllUnprivileged,
+                    // Will remove this when I figure out exactly what intents we need
+                    LogGatewayIntentWarnings = false,
+                    LogLevel = LogSeverity.Debug
+                });
 
-app.UseHttpsRedirection();
-app.UseStaticFiles();
+                return client;
+            });
+            // Interaction and command handling
+            builder.Services.AddSingleton(x => new InteractionService(x.GetRequiredService<DiscordSocketClient>()));
+            builder.Services.AddSingleton<InteractionHandler>();
+            
+            // Event hooks
+            builder.Services.AddSingleton<BotEvents>();
```

Comment:
I would make an interface for these, then register them. Dependent classes would reference `IInteractionHandler` and `IBotEvents` in the constructor. This is just generally a good practice to follow for various reasons.